src/main/java/io/github/darealturtywurty/turtychemistry/common/item/HotIngot.java

```diff
@@ -0,0 +1,74 @@
+package io.github.darealturtywurty.turtychemistry.common.item;
+
+import net.minecraft.nbt.CompoundTag;
+import net.minecraft.network.chat.Component;
+import net.minecraft.world.damagesource.DamageSource;
+import net.minecraft.world.entity.Entity;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.item.Item;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.item.TooltipFlag;
+import net.minecraft.world.level.Level;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.List;
+import java.util.concurrent.ThreadLocalRandom;
+
+public final class HotIngot extends ChemistryItem {
+
+    private float temperatureInCelsius;
+    private final Item defaultItem;
+    private static final String COMPOUND_TAG_ID = "temperature";
+    private static final int COOLING_COEFFICIENT = -4;
+
+    public HotIngot(final Builder builder, final Item nonHeatedUpVersion) {
+        super(builder);
+        defaultItem = nonHeatedUpVersion;
```

Comment:
I thought we agreed on it being the same item?

Reply:
right forgot about that, whoops